Gateway/Validator/PaymentLinksResponseValidator.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2022 Adyen N.V. (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Gateway\Validator;
+
+use Adyen\Payment\Helper\Data;
+use Adyen\Payment\Logger\AdyenLogger;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Payment\Gateway\Helper\SubjectReader;
+use Magento\Payment\Gateway\Validator\AbstractValidator;
+use Magento\Payment\Gateway\Validator\ResultInterface;
+use Magento\Payment\Gateway\Validator\ResultInterfaceFactory;
+
+class PaymentLinksResponseValidator extends AbstractValidator
+{
+    /**
+     * @var AdyenLogger
+     */
+    private $adyenLogger;
+
+    /**
+     * @var Data
+     */
+    private $adyenHelper;
+
+    /**
+     * CheckoutResponseValidator constructor.
+     *
+     * @param ResultInterfaceFactory $resultFactory
+     * @param AdyenLogger $adyenLogger
+     * @param Data $adyenHelper
+     */
+    public function __construct(
+        ResultInterfaceFactory $resultFactory,
+        AdyenLogger $adyenLogger,
+        Data $adyenHelper
+    ) {
+        $this->adyenLogger = $adyenLogger;
+        $this->adyenHelper = $adyenHelper;
+        parent::__construct($resultFactory);
+    }
+
+    /**
+     * @param array $validationSubject
+     * @return ResultInterface
+     */
+    public function validate(array $validationSubject)
+    {
+        $response = SubjectReader::readResponse($validationSubject);
+
+        $isValid = true;
+        $errorMessages = [];
+
+        if (!empty($response['error']) || $response['status'] !== 'active') {
+            $isValid = false;
+
+            if (!empty($response['error'])) {
+                $this->adyenLogger->error($response['error']);
+            }
+
+            $errorMsg = __('Error with payment method, please select a different payment method.');
+            throw new LocalizedException($errorMsg);
+        }
+
+        return $this->createResult($isValid, $errorMessages);
```

Comment:
Is it intended for `$errorMessages` to always be an empty array?

Reply:
In `CheckoutResponseValidator` we also have an empty array for `errorMessage`. I think it was added as a placeholder for possible additions in the past. Therefore, I wanted to keep the same structure. If we remove it from here, it also makes sense to remove it from `CheckoutResponseValidator` which is also unused.